docs/scripts/api/buildGridEventsDocumentation.ts

```diff
@@ -76,8 +77,35 @@ export default function buildGridEventsDocumentation(options: BuildEventsDocumen
     });
   });
 
+  const defaultProject = projects.get('x-data-grid-premium')!;
+
+  const dataGridPremiumProps = defaultProject.checker
+    .getTypeAtLocation(
+      (defaultProject.exports.DataGridPremiumProps.declarations![0] as ts.ExportSpecifier).name,
+    )
+    .getProperties();
+
+  dataGridPremiumProps.forEach((prop) => {
+    const declaration = prop.declarations?.find(ts.isPropertySignature);
+
+    if (
+      declaration?.type &&
+      ts.isTypeReferenceNode(declaration.type) &&
+      ts.isIdentifier(declaration.type.typeName) &&
+      declaration.type.typeName.escapedText === 'GridEventListener'
```

Comment:
We are not retrieving the handler of the control props here (`onFilterModelChange` for instance) which have another signature.

Not sure it would be a good idea to document them in the "Event" page, that might be confusing